### PR TITLE
fix(ProgressLogger): add a Manager to better handle extensibility

### PR DESCRIPTION
**Pull request by BotellaA**
There is an example in the test

### CODE REVIEW

File: include/geode/basic/progress_logger_manager.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        static void register_client(
+            std::unique_ptr< ProgressLoggerClient >&& client );
+
+        static void start( const std::string& message, index_t nb_steps );
+
+        static void update( index_t current_step, index_t nb_steps );
+
+        static void complete();
+
+        static void failed();
```
Comment (BotellaA) [inline include/geode/basic/progress_logger_manager.h:50]: Maybe these could be protected by a key so only the ProgressLogger can call them.

File: include/geode/basic/console_progress_logger_client.h [hunk 2]
```diff
@@ -35,15 +35,17 @@ namespace geode
...
     private:
-        void start( index_t nb_steps ) override;
+        void start( const std::string& message, index_t nb_steps ) override;
 
         void update( index_t current_step, index_t nb_steps ) override;
 
-        void end( index_t current_step, index_t nb_steps ) override;
+        void complete() override;
+
+        void failed() override;
```
Comment (panquez) [inline include/geode/basic/console_progress_logger_client.h:48]: What are the purpose of these two methods? Is it intentional that one use "ed" but not the other one?

Comment (BotellaA) [reply include/geode/basic/console_progress_logger_client.h:48]: The goal is to signal the end of the progress and how it ended.

Comment (BotellaA) [reply include/geode/basic/console_progress_logger_client.h:48]: There is probably a better choice of words 😜

Comment (panquez) [reply include/geode/basic/console_progress_logger_client.h:48]: Ok i would say `completed()` and `failed()`

Comment (BotellaA) [reply include/geode/basic/console_progress_logger_client.h:48]: Could you do the changes please?

**Pull request merged**